bindings/cpp/src/Session.cpp

```diff
@@ -439,8 +454,26 @@ Subscribe::~Subscribe()
     }
 }
 
+void Subscribe::check_custom_loop_options(sr_subscr_options_t opts)
+{
+    if (uses_custom_loop && !(opts & SR_SUBSCR_NO_THREAD)) {
+        throw std::logic_error("Custom loop subscription not called with SR_SUBSCR_NO_THREAD.");
+    }
+}
+
+void Subscribe::call_reg()
+{
+    if (uses_custom_loop && !reg_called) {
+        int pipe;
+        sr_get_event_pipe(ctx, &pipe);
+        reg(pipe, [this] {sr_process_events(ctx, sess->_sess, nullptr);});
+        reg_called = true;
+    }
+}
+
 void Subscribe::module_change_subscribe(const char *module_name, ModuleChangeCb cb, const char *xpath, uint32_t priority, sr_subscr_options_t opts)
 {
+    check_custom_loop_options(opts);
     module_change_cbs.emplace_back(cb);
 
     opts |= SR_SUBSCR_CTX_REUSE;
```

Comment:
I understand that's how the old code worked, but it's different from how `SR_SUBSCR_NO_THREAD` is enforced. My suggestion is to make this consistent, meaning to error out unless `SR_SUBSCR_CTX_REUSE` is passed by the user.